### ISSUE TITLE
S-58 Check 1686 - TSELNE separating TSSRON and ISTZNE

### ISSUE THREAD

**Issue by KlasOstergren-SMA** [iho-ohi/ENCWG-S58-ENC-Validation-Checks#37]
S-58 Check 1686 is reporting an error where a TSELNE is separating a TSSRON and an ISTZNE. In check 1687 a TSEZNE is accepted to separate a TSSRON and an ISTZNE.
Since a TSS Roundabout by definition is a circular traffic lane, I propose that check 1686 is amended to not report when a TSSRON is separating an ISTZNE similar to how check 1687 is constructed.

**Comment by TomRichardson6** [comment]
IC-ENC supports this change

**Comment by alvarosanuy** [comment]
TSSLNE and TSSZNE seem to be 'the same thing' encoded using different geometric primitives. In fact the 2 objects could be called a single name and have both, Line and Area geometric primitives.

Having said this, the key difference is that the center of a TSSRON can be a TSSZNE (because it's an area feature) but can't be a TSSLNE because it does not make sense. Based on this, the only use case to make a TSSLNE adjoin a TSSRON would be to use it to try to create a visual effect and surround the 'open' AP of the TSSRON with a solid magenta line (TSSLNE). This would be incorrect because if some of the sides of a TSSRON are not shared with a TSSLPT or ISTZNE then that would become the exterior boundary of the traffic separation scheme (TSS) and should be encoded using TSSBND instead (dashed magenta line).
ISTZNE have its own area boundary symbology (magenta T symbols) and shouldn't have another line feature (TSSLNE) on top. The boundary between adjoining features is visible already. Only TSSZNE makes sense as a 'boundary' feature for ISTZNE as it creates a wider separation corridor between features.
In short TSSZNE can share boundaries with TSSRON and ISTZNE but I do not see a use case for TSSLNE being used at the edge of neither TSSRON nor ISTZNE.
Checks 1686 as is, looks ok to me.

**Comment by JeffWootton** [comment]
According to S-4 - B-435.1(f), "Traffic in an inshore traffic zone is separated from traffic in the adjacent traffic lane by either a separation zone or a separation line".  The adjacent traffic lane may be a roundabout, as demonstrated in the Examples of Routeing Measures figure in INT1 - M (noting however that in this example the boundary between the inshore traffic zone and the roundabout is a traffic separation zone).  

Looking at Alvaro's argument, if we were to take this to its full conclusion this would mean that the second part of Check 1686 would need to be removed from the Check, as there is no requirement to encode a TSELNE between a TWRTPT and an ISTZNE because the magenta T symbols denoting the edge of the ISTZNE would be portrayed in this instance.  I disagree with this as the guidance in S-4 should take precedence and the portrayal in the ECDIS should not be a factor in this case.  Therefore, I think that, based on the guidance in S-4 - B-435.1(f) and UOC at clauses 10.2.1.2 and 10.2.1.3, the change as proposed by Klas should be made to Check 1686.

[As an additional comment, I think there is actually a bit of ambiguity in S-57 as, where an edge of a traffic lane (including a roundabout) is also the edge of the TSS it must be encoded using TSSBND (noting **not** as TSELNE); but as TSELNE or TSEZNE (with no TSSBND) if the limit of the TSS is defined as the line or area (zone) between a TSSLPT or TSSRON and an ISTZNE even though, by definition, **ISTZNE is not part of a TSS** (the guidance in the UOC at clauses 10.2.1.2 and 10.2.1.3 implies otherwise).  However, I think this is a discussion to be had more in relation to S-101 as if this thinking is correct the changes required in S-57 and S-58 would result in invalidating related ENCs.]

**Comment by TomRichardson6** [comment]
I support the comments made by @JeffWootton and note that as some Routeing Measures are established by IMO we should be cautious about making any changes to the modelling here without strong reasons. With my IC-ENC hat on we often accept these warnings as the complexity of the real world situation means that they cant be applied in practice. the routes on the coast of the Netherlands would be an example I'd reference where we have a very complex mix of routeing measures of all varieties. Support raising an S-101 Issue for this in the relevant github @JeffWootton ?

**Comment by KlasOstergren-SMA** [comment]
Sorry for duplicating the comment from @JeffWootton. I wrote but missed to save my comment and realized this and saved without reading the new comments.

**Comment by alvarosanuy** [comment]
By definition (DCEG), the feature Separation Zone or Line must be used to encode the common boundary or separation areas between:

- two traffic lanes, or
- of one traffic lane and one inshore traffic zone, or
- **to encode the centre part of a roundabout**.

There's nothing about TSELNE being used as:

1. the boundary between TSSRON and ISTZNE (I think there are merits to allow this **but**, the definition in the UOC and DCEG needs to be reviewed) or

2. the edge (exterior limit) of a roundabout. I already explained it does not make sense to use it in its center. either by itself or around a TSEZNE if encoded.

**TSELNE**.....................................................................................**TSSBND**.......................................................................**TSSRON with TSEZNE in the middle**
![image](https://user-images.githubusercontent.com/70189016/229030749-43f1c1bc-e5cf-49e7-8252-04b345673036.png)

**Comment by HermanSchouten** [comment]
There seem to be quite some cases where error 1686/1687 is unjustly reported.
We have quite some Routing measures in NL area, the error reports for these areas cause quite some confusion.
Apart from the Deep water routes, a TSEZNE on the edge of an ENC cell will trigger the error, because there is nothing on that limit of the TSEZNE. 

DWRTPT and DWRLPT should probably be included in these checks.

Screenshots from IMO SHIPS’ ROUTEING: DWRTPT in the middle and a TSEZNE on either side and a TSSLP on the outer limits.
![20230912_110821](https://github.com/iho-ohi/ENCWG-S58-ENC-Validation-Checks/assets/90600414/56fe3839-6a63-4c6a-89b4-1b8c8cac0283)

Screenshot with ISTZNEs, these are defined in IMO SHIPS’ ROUTEING as well. Eastern part bounded by TSELNE  and in the Western part bounded by TSEZNE.
![20230912_111105](https://github.com/iho-ohi/ENCWG-S58-ENC-Validation-Checks/assets/90600414/834f26bf-c1ba-4d5e-9255-b42082fe55ab)

**Comment by Richard-Fowle** [comment]
Changes approved in principle at ENCWG8 final wording agreed by S58SubWG members included in S-58 edition 8.0.0 for submission to HSSC.
